mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -0,0 +1,167 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.core.User;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.FetchGroupImpl;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.model.MailboxMetaData;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.mailbox.model.search.MailboxQuery;
+import org.apache.james.mailbox.store.StoreMessageManager;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.util.OptionalUtils;
+import org.apache.james.util.streams.Iterators;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class DefaultMailboxBackup implements MailboxBackup {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultMailboxBackup.class);
+
+    protected class MailAccountContent {
+        private final MailboxWithAnnotations mailboxWithAnnotations;
+        private final Stream<MessageResult> messages;
+
+        MailAccountContent(MailboxWithAnnotations mailboxWithAnnotations, Stream<MessageResult> messages) {
+            this.mailboxWithAnnotations = mailboxWithAnnotations;
+            this.messages = messages;
+        }
+
+        public MailboxWithAnnotations getMailboxWithAnnotations() {
+            return mailboxWithAnnotations;
+        }
+
+        public Stream<MessageResult> getMessages() {
+            return messages;
+        }
+    }
+
+    public DefaultMailboxBackup(MailboxManager mailboxManager, ArchiveService archiveService, MailArchivesLoader archiveLoader, BlobStore store) {
+        this.mailboxManager = mailboxManager;
+        this.archiveService = archiveService;
+        this.archiveLoader = archiveLoader;
+        this.store = store;
```

Comment:
Please name this field `blobStore`... I just got confused about wich kind of store it was...